### PR TITLE
Add more info to installation errors

### PR DESCRIPTION
**Pull request by tushevorg**
Shows which file cannot be copied. It allows to fix issues when the target file was made read-only and thus cannot be overwritten, causing installation failure.

### CODE REVIEW

**Comment by zero-24** [comment]
@tushevorg  can you add some testinstructions how we can test this?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8136">issues.joomla.org/joomla-cms/8136</a>.</sub>

**Comment by tushevorg** [comment]
1. make any of the files inaccessible to PHP's copy (chmod 444 some-file.css for instance)
2. Upload update package that will replace the read-only file
3. You will receive informative error - which file failed to be copied, not the general "Something failed" error, so you will remember that you've chmod'ed the file half a year ago for some reason and will be able quickly resolve the issu<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8136">issues.joomla.org/joomla-cms/8136</a>.</sub>

File: libraries/joomla/filesystem/folder.php [hunk 1]
```diff
@@ -309,8 +310,15 @@ public static function delete($path)
...
-		// Check to make sure the path valid and clean
-		$path = $pathObject->clean($path);
+		try
+		{
+			// Check to make sure the path valid and clean
+			$path = $pathObject->clean($path);
+		}
+		catch (UnexpectedValueException $e)
+		{
+			throw $e;
```
Comment (mbabker) [inline libraries/joomla/filesystem/folder.php]: Can you revert this change?  A try/catch that just catches and re-throws an Exception without any processing is the same as not catching Exceptions at all.

**Comment by tushevorg** [comment]
Reverted, and added one more notification (as well in ftp)

**Comment by tushevorg** [comment]
Any updates?

**Comment by conconnl** [comment]
Hi,

Can you create a full test scenario including test package.
With the information Before Patch and After patch

Then I will test it and if successful I will ask someone else to test this.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8136">issues.joomla.org/joomla-cms/8136</a>.</sub>

**Comment by tushevorg** [comment]
Hi, @conconnl 

I'm not very sure about what are you asking for. Could you please explain?

The scenario looks like:
1. Install any extension, for instance http://extensions.joomla.org/extensions/extension/style-a-design/personal-layout/blank-component
2. Make any file, like controller.php, read-only (on Unix `chmod 444 controller.php`)
3. Install the Extension again, to simulate an update. Due to read-only restrictions, copying the new file will fail, and an error will raise.
4. This error without the patch will be just the general "Copy file failed" error. With the patch, it will contain the name of the file as well. 

It's all very simple, just a slight modification of the error message, but extremely useful for end-users.

**Comment by conconnl** [comment]
I have tested this and i don't see any difference.
So I think you didn't test it for a while on new Joomla! versions.

Before Patch:
screen shot 2016-06-30 at 00 46 07

After Patch:
screen shot 2016-06-30 at 00 47 26

**Comment by hardiktailored** [comment]
As @conconnl said above I find no difference applying patch. File name already showing without patch too.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8136">issues.joomla.org/joomla-cms/8136</a>.</sub>

**Comment by brianteeman** [comment]
I am closing this at this time as it would appear to have been fixed elsewhere. If that is not correct then it can always be reopened<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8136">issues.joomla.org/joomla-cms/8136</a>.</sub>
